### PR TITLE
feat: Add charge link event publisher project

### PR DESCRIPTION
**Pull request by bemwen**
## Description

The purpose of this PR is to add a simple .NET 5.0 based service bus trigger that will evolve into the charge link event publisher.

Only the most basic functionality has been added. The rest, and pipelines, will be added in a separate PR later

* .NET 5.0 project added
* Simple service bus trigger added (just logs receiving events for now)
* HealthStatus added, as per our normal usage
* Configuration matching terraform deployment added

## References

* #359

### CODE REVIEW

**Review by Mech0z — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/GreenEnergyHub.Charges.ChargeLinkChargeLinkEventPublisher.csproj [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/GreenEnergyHub.Charges.ChargeLinkChargeLinkEventPublisher.csproj:18]: <Nullable>enable</Nullable>
<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
<LangVersion>9</LangVersion>

Maybe all of the "link" projects are missing these projects settings?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/Program.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+            ConfigureMessaging(serviceCollection);
+        }
+
+        private static void ConfigureMessaging(IServiceCollection serviceCollection)
+        {
+            serviceCollection.AddScoped<ICorrelationContext, CorrelationContext>();
+        }
+
+        private static string GetEnv(string variableName)
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/Program.cs:46]: unused?

**Review by lasrinnil — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/Program.cs (hunk 2 above)
Comment (bemwen) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/Program.cs:46]: For now, not shortly

Comment (lasrinnil) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/Program.cs:46]: should unused methods not be a warning? And aren't we treating warnings as erros?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/GreenEnergyHub.Charges.ChargeLinkChargeLinkEventPublisher.csproj (hunk 1 above)
Comment (bemwen) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ChargeLinkEventPublisher/GreenEnergyHub.Charges.ChargeLinkChargeLinkEventPublisher.csproj]: Added to this and receiver link project

**Pull request merged**